### ISSUE TITLE
External USB-C monitors not detected automatically (but rescan does find them)

### ISSUE THREAD

**Issue by NBPrA** [emoacht/Monitorian#372]
**Summary**
When changing monitors (connected via USB-C), the newly plugged in monitor is not detected (but performing a "Rescan monitors" does then detect them). I presume it is expected that Monitorian automatically detects the new monitor on USB-C / monitor state change.

**Hardware **
Dell Lattitude 5420
ASUS PA24AC
HP Z24i G2 + Plugable UD-CAM

**Behaviour**
Observed behaviour:

1. Laptop plugged in to monitor A
2. Monitorian detects monitor A
3. Unplug USB-C cable from Monitor A. Plug in to monitor B.
4. Monitorian does not automatically monitor B
5. Perform "Rescan monitors" in Monitorian
6. Monitorian detects monitor B.

When swapping back to monitor A, the same behaviour is seen.

Expected behaviour:

At step 4, Monitorian detects monitor B automatically.

**Files:**
[probe-PA24AC.log](https://github.com/emoacht/Monitorian/files/9951039/probe-PA24AC.log)
[operation.log](https://github.com/emoacht/Monitorian/files/9951040/operation.log)
[probe-Z24.log](https://github.com/emoacht/Monitorian/files/9951041/probe-Z24.log)

Thanks.

**Comment by emoacht** [comment]
Thank you for reporting in details.
Could you give me the time exactly when following events happens? Otherwise, I cannot check whether any event is recorded at the time of those events.

> Unplug USB-C cable from Monitor A. Plug in to monitor B.
> Monitorian does not automatically monitor B
> Perform "Rescan monitors" in Monitorian

If you didn't recorded them, could you retry this sequence again and post the time of each event with the renewed operation.log?

**Comment by emoacht** [comment]
Thanks for the log and time.

According the log, Z24i was detected normally by ScanAsync at 10/11/2022 11:57:14. At that time Z24i should have been shown normally. Did you watch it?

**Comment by emoacht** [comment]
Thanks again.
Then your PC's power is supplied by USB-C from the monitor along with video signals?

**Comment by emoacht** [comment]
I rechecked the internal watchers and operation.log.
As your PC is powered from a monitor, the reconnection will be notified by the OS as changes both in power status and display settings. Thus it will be handled by both power watcher and display watcher.

I found what happened from operation.log.
13:49:22 The reconnection was notified by the OS. This is not recorded in log.
13:49:23 Both power watcher and display watcher triggered first rescan (1 sec after notification). PA24A was found as uncontrollable.
13:49:27 Both power watcher and display watcher triggered second rescan (5 sec after notification). PA24A was found as uncontrollable again.
13:49:32 Display watcher triggered third (last) rescan (10 sec after notification). PA24A was found as controllable. Then update brightness was called but failed. This failure was recorded on 13:49:35. If update brigntness succeeded, PA24A would pass rescan and be shown. But it failed and was kept unshown.

In short, at the time of 10 sec after reconnection, PA24A reported the capabilities for brightness but actually this app failed to get brightness. In other words, 10 sec was too short for PA24A to return to normal state.

**Comment by emoacht** [comment]
Added another rescan (20 sec after notification) by 3f5182ade89b3d2d2c2b4de246c477d7a000d60c
